### PR TITLE
Add implementation of SQLDataSources

### PR DESCRIPTION
**Pull request by detule**
Hi @lexicalunit @mloskot,

This is an implementation of [SQLDataSources](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqldatasources-function?view=sql-server-ver15).

* The implementation is very similar to the existing `list_drivers`.
* In an attempt to add a test, I ended up touching quite a bit of the CI scripting - let me know if you have any suggestions on a better way of doing this.

Thank you!

### CODE REVIEW

File: nanodbc/nanodbc.cpp [hunk 1]
```diff
@@ -3439,6 +3439,59 @@ void result::result_impl::get_ref_impl(short column, T& result) const
 namespace nanodbc
 {
 
+std::list<datasource> list_dsns()
```
Comment (mloskot) [inline nanodbc/nanodbc.cpp:3442]: I'd suggest we avoid abbreviations and use clear self-descriptive names, i.e. `list_datasources`

File: test/test_case_fixture.h [hunk 2]
```diff
@@ -950,6 +950,22 @@ struct test_case_fixture : public base_test_fixture
         REQUIRE(found);
     }
 
+    void test_dsns()
```
Comment (mloskot) [inline test/test_case_fixture.h:953]: similarly, `test_datasources`

Comment (detule) [reply test/test_case_fixture.h:953]: Thanks, will change.

File: nanodbc/nanodbc.cpp (hunk 1 above)
Comment (detule) [reply nanodbc/nanodbc.cpp:3442]: Thanks, will change.

Comment (detule) [reply nanodbc/nanodbc.cpp]: Done - let me know if I missed anything.

File: test/test_case_fixture.h (hunk 2 above)
Comment (detule) [reply test/test_case_fixture.h]: Done - let me know if I missed anything.

File: nanodbc/nanodbc.cpp (hunk 1 above)
Comment (mloskot) [reply nanodbc/nanodbc.cpp]: Thanks

**Comment by lexicalunit** [comment]
Very cool, thanks @detule!

**Pull request merged**